userspace/libscap/engine/bpf/scap_bpf.c

```diff
@@ -797,21 +797,18 @@ static int load_all_progs(struct bpf_engine *handle) {
 	return SCAP_SUCCESS;
 }
 
+// should fit one full event (given that length is 16 bits)
+#define RINGBUFFER_SLACK (1 << 16)
 static void *perf_event_mmap(struct bpf_engine *handle,
                              int fd,
                              unsigned long *size,
                              unsigned long buf_bytes_dim) {
 	int page_size = getpagesize();
 	unsigned long ring_size = buf_bytes_dim;
 	int header_size = page_size;
-	unsigned long total_size = ring_size * 2 + header_size;
+	unsigned long total_size = ring_size + header_size + RINGBUFFER_SLACK;
```

Comment:
You explained that in the commit message, but maybe it's worth mentioning it here